### PR TITLE
feat: expose SimpleEditorService

### PR DESCRIPTION
**Pull request by CompuIves**
Here I am again! Getting closer to implementing our own tab system, but we decided that we will use Monaco editors first, inside the tabs. For this, we need to return an EditorPane, and I saw that you already built this nice wrapper which exposes `EditorPane`. I initially tried copying it to our codebase, but there are some functions inside VSCode that we cannot import (`isPreferredGroup`, `SIDE_GROUP`). I thought this would be easier for us, but let me know if you prefer it to be handled differently.

### CODE REVIEW

**Comment by CGNonofr** [comment]
What about exposing the `EditorService` instead, so you can override it?

Btw, what feature is missing from it that requires you to implement your custom implementation?

**Comment by CompuIves** [comment]
Good one, that works too. I could even use the `openEditor` of that service.

> Btw, what feature is missing from it that requires you to implement your custom implementation?

No specific feature missing, what I'm trying to do is quite custom. This is our editor:
image

As you can see we already have a tab system (with tab splitting, etc...). I'm looking if we can implement a custom EditorGroupsService that will work with _our_ tab system. So that other services like `IHistoryService` can interact with our tab system. This remains an experiment (how easy is it to do?). If it doesn't work, I'm considering replacing our editor pane with the VSCode editor part.

File: src/service-override/editor.ts [hunk 1]
```diff
@@ -98,5 +98,6 @@ export {
   OpenEditor,
   IEditorOptions,
   IResolvedTextEditorModel,
-  IReference
+  IReference,
+  EditorService
```
Comment (CGNonofr) [inline src/service-override/editor.ts:102]: If it's exposed, we should probably give it a less generic name, like `SimpleEditorService` or something, don't you think?

Comment (CompuIves) [reply src/service-override/editor.ts:102]: Good one, will do!

Comment (CGNonofr) [reply src/service-override/editor.ts]: Just change the commit message and we're good to go :)

**Comment by CGNonofr** [comment]
Oh there is some warnings, as the class is exported, it needs the return types and parameters to be explicit

also if you want to be able to override methods, you need the class to repeat the parameters (prefixed by a `_` to not have a warning because unused)

**Comment by CompuIves** [comment]
Fixed the TS case, but couldn't get the params to work (it still complains):
image

For me, I mainly need access to `wrapOpenEditor`, as I need to call it from `openEditor` in an `EditorGroup`. An alternative is that I use `editorService` directly and call `openEditor` on that one, but for that to work I will need to add two more events to `editorService` (`onDidEditorOpenFail` and another one), because the `IHistoryService` expects that for some reason.

So for me merging this will be already enough, or exposing `wrapOpenEditor` separately or adding the two events to `editorService`. Do you have a preference?

**Comment by CGNonofr** [comment]
The issue was due to the eslint rule not being on main yet but on an open PR

I've put the fix in [another PR](https://github.com/CodinGame/monaco-vscode-api/pull/142)

**Comment by CGNonofr** [comment]
Are you ok with closing that PR? Are the changes fine?

**Comment by CompuIves** [comment]
Ahh, thanks a bunch for doing the work on this! Absolutely fine with closing the PR.

**Pull request closed without merging**